From 9da8679e8f716e99ed4618fff657033fe4d102bc Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Wed, 18 Jan 2023 22:28:05 +0000
Subject: [PATCH] fix build with 64-bit time_t

:Release Notes:
Use input_event_sec/input_event_usec to access sec and usec
from input_event.

:Detailed Notes:
input_event is defined like this:
  /*
   * The event structure itself
   * Note that __USE_TIME_BITS64 is defined by libc based on
   * application's request to use 64 bit time_t.
   */

  struct input_event {
  #if (__BITS_PER_LONG != 32 || !defined(__USE_TIME_BITS64)) && !defined(__KERNEL__)
          struct timeval time;
  #define input_event_sec time.tv_sec
  #define input_event_usec time.tv_usec
  #else
          __kernel_ulong_t __sec;
  #if defined(__sparc__) && defined(__arch64__)
          unsigned int __usec;
          unsigned int __pad;
  #else
          __kernel_ulong_t __usec;
  #endif
  #define input_event_sec  __sec
  #define input_event_usec __usec
  #endif
          __u16 type;
          __u16 code;
          __s32 value;
  };

using the macros allows to support __USE_TIME_BITS64 with backwards
compatibility.

Fixes:
http://gecko.lge.com:8000/Errors/Details/524108

build with time64.inc as in:
https://git.openembedded.org/openembedded-core/tree/meta/conf/distro/include/time64.inc?id=18df5694796bcfee73c3765bc991bcef055466e3

TOPDIR/BUILD/work/qemux86-webos-linux/qtbase-plugins-webos/1.0.0-14-r5/git/src/eglfs_webos/emulator/qemulatorkeyboardhandler.cpp: In member function 'void QEmulatorKeyboardHandler::switchLed(int, bool)':
TOPDIR/BUILD/work/qemux86-webos-linux/qtbase-plugins-webos/1.0.0-14-r5/git/src/eglfs_webos/emulator/qemulatorkeyboardhandler.cpp:126:28: error: 'struct input_event' has no member named 'time'
  126 |     ::gettimeofday(&led_ie.time, 0);
      |                            ^~~~

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRP-310] Apply .patch files for hot-fix to corresponding repositories
[WRP-11630] Add Upstream-Status to existing .patch files in meta-lg-webos
[WRP-11864] CCC: Various build fixes and .patch files cleanup
[WRP-7187] Fix build with 64-bit time_t

Change-Id: I199e4f9b90de3b7aa82a2895caa86b214aab0ca7
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/qtbase-plugins-webos/+/349068 fix build with 64-bit time_t]

 src/eglfs_webos/emulator/qemulatorkeyboardhandler.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/eglfs_webos/emulator/qemulatorkeyboardhandler.cpp b/src/eglfs_webos/emulator/qemulatorkeyboardhandler.cpp
index 0073066..2cf469e 100644
--- a/src/eglfs_webos/emulator/qemulatorkeyboardhandler.cpp
+++ b/src/eglfs_webos/emulator/qemulatorkeyboardhandler.cpp
@@ -123,7 +123,10 @@ void QEmulatorKeyboardHandler::switchLed(int led, bool state)
     qWarning() << "switchLed" << led << state;
 
     struct ::input_event led_ie;
-    ::gettimeofday(&led_ie.time, 0);
+    struct ::timeval curtime;
+    ::gettimeofday(&curtime, 0);
+    led_ie.input_event_sec = curtime.tv_sec;
+    led_ie.input_event_usec = curtime.tv_usec;
     led_ie.type = EV_LED;
     led_ie.code = led;
     led_ie.value = state;
